common/oauthTokenManager.go

```diff
@@ -713,16 +724,134 @@ func (credInfo *OAuthTokenInfo) GetNewTokenFromMSI(ctx context.Context) (*adal.T
 	if credInfo.IdentityInfo.MSIResID != "" {
 		params.Set("msi_res_id", credInfo.IdentityInfo.MSIResID)
 	}
+
 	req.URL.RawQuery = params.Encode()
 	req.Header.Set("Metadata", "true")
+
 	// Set context.
 	req.WithContext(ctx)
 
 	// Send request
+	msiTokenHTTPClient.Timeout = 10 * time.Second
 	resp, err := msiTokenHTTPClient.Do(req)
+	msiTokenHTTPClient.Timeout = 0
+	return req, resp, err
+}
+
+// checkIfWWWAuthenticateUnavailable checks if the key "Www-Authenticate" is unavailable in the header of an http response
+func checkIfWWWAuthenticateUnavailable(resp *http.Response) bool {
```

Comment:
A more generic name like isValidArcResponse() would be better.
Inside that you can search for this specific header explaining with a comment why this header qualified for a valid Arc response.